src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -730,19 +744,23 @@ CHIP_ERROR Encode(chip::TLV::TLVWriter & writer, chip::TLV::Tag tag) const
 
 template <typename DecodableValueType> class BufferedReadClientCallback final : public app::ReadClient::Callback {
 public:
-    using OnSuccessCallbackType
-        = std::function<void(const app::ConcreteClusterPath & aPath, const uint32_t aValueId, const DecodableValueType & aData)>;
+    using OnSuccessAttributeCallbackType = std::function<void(
+        const app::ConcreteClusterPath & aPath, const AttributeId aAttributeId, const DecodableValueType & aData)>;
```

Comment:
```suggestion
        const ConcreteAttributePath & aPath, const DecodableValueType & aData)>;
```
if we're going back to having separate callbacks for events and attributes, we can just use the right type.